packages/andromeda-automation/src/oracle.rs

```diff
@@ -0,0 +1,60 @@
+use common::ado_base::{AndromedaMsg, AndromedaQuery};
+use cosmwasm_schema::{cw_serde, QueryResponses};
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    // The contract which we'll query data from
+    pub target_address: String,
+
+    // The query message's binary
+    pub message_binary: String,
```

Comment:
This can be `Binary`, when sending from frontend sending a base64 encoded string is valid, when sending from another contract you can just use `to_binary`.